pkg/utils/maps/maps.go

```diff
@@ -19,6 +19,16 @@ func IsSubset(toCheck, fullSet map[string]string) bool {
 	return true
 }
 
+// IsEqualSubset compares the values of the given subset of keys between two maps and returns true if they are not equal.
```

Comment:
nit: it feels a bit odd to me that a function named `IsEqual` returns `true` when things are `not` equal.

Reply:
Oh yes we need to change that. Or rename it so that it becomes clear.

Reply:
sorry for the oversight I will fix this.